Tests/FluentAssertions.Specs/Exceptions/AsyncFunctionExceptionAssertionSpecs.cs

```diff
@@ -1187,6 +1187,48 @@ public async Task When_no_exception_should_be_thrown_on_UI_thread_for_async_func
             // Assert
             await act.Should().NotThrowAsync();
         }
+
+        [Fact]
+        public async Task When_async_method_throws_the_exact_expected_inner_exception_it_should_succeed()
+        {
+            // Arrange
+            Func<Task> task = async () =>
+            {
+                await Task.Delay(100);
+                throw new AggregateException(new InvalidOperationException());
+            };
+
+            var s = await task.Should().ThrowAsync<AggregateException>();
+
+            s.WithInnerExceptionExactly<InvalidOperationException>();
+
+            // Act
+            Func<Task> action = () => task
+                .Should().ThrowAsync<AggregateException>()
+                .WithInnerExceptionExactly<AggregateException, InvalidOperationException>();
+
+            // Assert
+            await action.Should().NotThrowAsync();
+        }
+
+        [Fact]
+        public async Task When_async_method_does_not_throw_the_expected_exact_inner_exception_it_should_fail()
+        {
+            // Arrange
+            Func<Task> task = async () =>
+            {
+                await Task.Delay(100);
+                throw new AggregateException(new ArgumentException());
+            };
+
+            // Act
+            Func<Task> action = () => task
+                .Should().ThrowAsync<AggregateException>()
+                .WithInnerExceptionExactly<AggregateException, InvalidOperationException>();
```

Comment:
If you want to verify that `because` and `becauseArgs` are properly propagated, you can get inspiration from `Should_fail_with_descriptive_message_when_asserting_boolean_value_not_to_be_equal_to_a_different_value`.

Reply:
That was debug as well. I removed it.